app/policies/facility_policy.rb

```diff
@@ -0,0 +1,65 @@
+class FacilityPolicy < ApplicationPolicy
+  def index?
+    user && (user.superuser? || user.secondary_nurse? || user.medical_officer?)
+  end
+
+  def show?
+    user && (user.superuser? || user.facility_id == record.id)
+  end
+
+  def new?
+    user && user.superuser?
+  end
+
+  def create?
+    user && user.superuser?
+  end
+
+  def show_users?
+    user && (user.superuser? || user.facility_id == record.id)
+  end
+
+  def show_patients?
+    user && (user.superuser? || user.facility_id == record.id)
+  end
+
+  def assign_facility?
+    if user
+      if user.superuser?
+        return true
+      elsif user.medical_officer?
+        return user.facility.id == record.id
+      else
+        return false
+      end
+    else
+      return false
+    end
+    # binding.pry
+    # return true
+  end
```

Comment:
This is more like writing Js. The ruby way would be
```suggestion
  def assign_facility?
     return if user.blank?
     
     return true if user.superuser?

     return false unless user.medical_officer?

     user.facility.id == record.id
  end
```